improver/lapse_rate.py

```diff
@@ -455,33 +456,33 @@ def process(self, temperature, orography, land_sea_mask,
 
         # Create list of arrays over "realization" coordinate
         has_realization_dimension = False
+        original_dimension_order = None
         if temperature_cube.coords("realization", dim_coords=True):
+            original_dimension_order = get_dim_coord_names(temperature_cube)
             enforce_coordinate_ordering(temperature_cube, "realization")
-            temp_slices = temperature_cube.data
+            temp_data_slices = temperature_cube.data
             has_realization_dimension = True
         else:
-            temp_slices = [temperature_cube.data]
+            temp_data_slices = [temperature_cube.data]
 
         # Calculate lapse rate for each realization
-        lapse_rate_data = None
-        for temperature_data in temp_slices:
+        lapse_rate_data = []
+        for temperature_data in temp_data_slices:
             lapse_rate_array = self._generate_lapse_rate_array(
                 temperature_data, orography_data, land_sea_mask_data)
-
-            if lapse_rate_data is None:
-                lapse_rate_data = lapse_rate_array
-                if has_realization_dimension:
-                    lapse_rate_data = [lapse_rate_data]
-            elif not isinstance(lapse_rate_data, list):
-                lapse_rate_data = [lapse_rate_data]
-                lapse_rate_data.append(lapse_rate_array)
-            else:
-                lapse_rate_data.append(lapse_rate_array)
+            lapse_rate_data.append(lapse_rate_array)
+        lapse_rate_data = np.array(lapse_rate_data, dtype=np.float32)
```

Comment:
Is the dtype specification necessary here? I don't think Numpy changes the dtype of data unless a mathematical operation is applied to it.

Reply:
Actually probably not.  I've tidied this up in a couple of places here.